### PR TITLE
fix: better log on informer connection error

### LINKED ISSUE #6138: improve traces for informer
### Is your enhancement related to a problem? Please describe

The traces are not really helpful for an end-user

I get for example after starting Podman Desktop:
```
11:10:11 AM [main] informer start error:  HttpError: HTTP request failed
11:10:25 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:60601
11:10:25 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:44829
11:10:28 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:60601
11:10:29 AM [main] informer start error:  HttpError: HTTP request failed
11:10:31 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:44829
11:10:34 AM [main] informer start error:  HttpError: HTTP request failed
11:11:03 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:60601
11:11:05 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:44829
11:11:09 AM [main] informer start error:  HttpError: HTTP request failed
11:11:11 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:60601
11:11:15 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:44829
11:11:20 AM [main] informer start error:  HttpError: HTTP request failed
11:12:20 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:60601
11:12:23 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:44829
11:12:28 AM [main] informer start error:  HttpError: HTTP request failed
11:12:35 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:60601
11:12:44 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:44829
11:12:52 AM [main] informer start error:  HttpError: HTTP request failed
11:13:36 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:60601
11:13:41 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:44829
11:13:48 AM [main] informer start error:  HttpError: HTTP request failed
11:14:00 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:60601
11:14:13 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:44829
11:14:24 AM [main] informer start error:  HttpError: HTTP request failed
11:14:53 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:60601
11:14:59 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:44829
11:15:07 AM [main] informer start error:  HttpError: HTTP request failed
11:15:24 AM [main] informer start error:  Error: connect ECONNREFUSED 127.0.0.1:60601
```

### Describe the solution you'd like

almost no-one knows what would be an informer

I would expect the trace should say something more helpful like

`Trying to reach the kubernetes cluster defined in /Users/benoitf/kube/.config named "minikube"  at "127.0.0.1:44829" but got a connection refused, retrying the connection in 2mn...`

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What issues does this PR fix or reference?

Fixes #6138 

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/main/src/plugin/kubernetes-context-state.ts [hunk 1]
```diff
@@ -291,16 +297,19 @@ export class ContextsManager {
     options: CreateInformerOptions<T>,
   ) {
     informer.start().catch((err: unknown) => {
+      const nextTimeout = options.backoff.get();
       if (err !== undefined) {
-        console.warn('informer start error: ', String(err));
+        console.warn(
+          `Trying to watch ${options.resource} on the kubernetes context named "${context.name}" but got a connection refused, retrying the connection in ${nextTimeout / 1000}s. ${String(err)})`,
```
Comment (axel7083) [inline packages/main/src/plugin/kubernetes-context-state.ts]: Aren't we gonna have the console spammed with those messages when our cluster is not started (E.g. Kind) ?

Comment (feloy) [reply packages/main/src/plugin/kubernetes-context-state.ts]: At startup, you'll get spammed with these logs, but with the backoff retry, you'll get a message every minute only after a couple of minutes

Comment (axel7083) [reply packages/main/src/plugin/kubernetes-context-state.ts]: Maybe we should use debug ? So if user is trying to find an error in the logs, he do not only see hundreds of those messages after a few hours of running. I feel like, if this is expected to fail, we do not need to log it 

What do you think ?

Comment (feloy) [reply packages/main/src/plugin/kubernetes-context-state.ts]: ok, let's change to debug

**Review by axel7083 — APPROVED** [review]
LGTM! 🚀

**Review by benoitf — COMMENTED** [review]
```
11:55:38 AM [main] Trying to watch pods on the kubernetes context named "minikube" but got a connection refused, retrying the connection in 9.852s. Error: connect ECONNREFUSED 127.0.0.1:44829)

11:55:38 AM [main] Trying to watch deployments on the kubernetes context named "minikube" but got a connection refused, retrying the connection in 10.307s. Error: connect ECONNREFUSED 127.0.0.1:44829)

11:55:38 AM [main] Trying to watch pods on the kubernetes context named "kind-kind-cluster" but got a connection refused, retrying the connection in 10.176s. Error: connect ECONNREFUSED 127.0.0.1:60601)

11:55:39 AM [main] Trying to watch pods on the kubernetes context named "fbenoit-dev/api-sandbox-m3-1530-p1-openshiftapps-com:6443/fbenoit" but got a connection refused, retrying the connection in 10.446s. HttpError: HTTP request failed)

11:55:39 AM [main] Trying to watch deployments on the kubernetes context named "fbenoit-dev/api-sandbox-m3-1530-p1-openshiftapps-com:6443/fbenoit" but got a connection refused, retrying the connection in 10.697s. HttpError: HTTP request failed)

11:55:48 AM [main] Trying to watch deployments on the kubernetes context named "kind-kind-cluster" but got a connection refused, retrying the connection in 20.197s. Error: connect ECONNREFUSED 127.0.0.1:60601)

11:55:48 AM [main] Trying to watch pods on the kubernetes context named "minikube" but got a connection refused, retrying the connection in 19.847s. Error: connect ECONNREFUSED 127.0.0.1:44829)

11:55:48 AM [main] Trying to watch pods on the kubernetes context named "kind-kind-cluster" but got a connection refused, retrying the connection in 20.638s. Error: connect ECONNREFUSED 127.0.0.1:60601)

11:55:48 AM [main] Trying to watch deployments on the kubernetes context named "minikube" but got a connection refused, retrying the connection in 20.635s. Error: connect ECONNREFUSED 127.0.0.1:44829)

11:55:49 AM [main] Trying to watch pods on the kubernetes context named "fbenoit-dev/api-sandbox-m3-1530-p1-openshiftapps-com:6443/fbenoit" but got a connection refused, retrying the connection in 20.992s. HttpError: HTTP request failed)

11:55:50 AM [main] Trying to watch deployments on the kubernetes context named "fbenoit-dev/api-sandbox-m3-1530-p1-openshiftapps-com:6443/fbenoit" but got a connection refused, retrying the connection in 21.644s. HttpError: HTTP request failed)

```

tried, thanks for the enhancement, not sure if we need the values after the dot on timers

`20.635s` -> `20s`

unit tests are failing

**Review by benoitf — APPROVED** [review]

**Pull request merged**